File: core/src/commonMain/kotlin/com/xhsdn/core/http/DownloadCallback.kt
package com.xhsdn.core.http

/**
 * 下载进度回调。
 * 对应 [app/.../DownloadCallback.java]，行为完全一致。
 */
interface DownloadCallback {
    /** 文件下载成功 */
    fun onFileDownloaded(filePath: String)
    /** 单个文件下载失败 */
    fun onDownloadError(status: String, originalUrl: String)
    /** 文本进度（如 "Live photo 合成失败"） */
    fun onDownloadProgress(status: String)
    /** 字节级进度 */
    fun onDownloadProgressUpdate(downloaded: Long, total: Long)
    /** 检测到视频（用于触发 "是否继续" 提示） */
    fun onVideoDetected()
    /** 是否已取消 */
    fun isCancelled(): Boolean
}

File: web/src/platform/DownloadCallback.ts
/**
 * 下载回调。Web 版默认是 no-op + console.log。
 * 后续要接 SSE / WebSocket 给前端实时进度时,改这里的实现即可。
 */
export interface DownloadCallback {
  onFileDownloaded(filePath: string): void;
  onDownloadError(status: string, originalUrl: string): void;
  onDownloadProgress(status: string): void;
  onDownloadProgressUpdate(downloaded: number, total: number): void;
  onVideoDetected(): void;
  isCancelled(): boolean;
}

export class ConsoleDownloadCallback implements DownloadCallback {
  onFileDownloaded(filePath: string): void {
    console.log(`[file] ${filePath}`);
  }
  onDownloadError(status: string, originalUrl: string): void {
    console.log(`[err] ${status} (${originalUrl})`);
  }
  onDownloadProgress(status: string): void {
    console.log(`[progress] ${status}`);
  }
  onDownloadProgressUpdate(downloaded: number, total: number): void {
    // no-op
  }
  onVideoDetected(): void {
    console.log('[video] 检测到视频');
  }
  isCancelled(): boolean {
    return false;
  }
}
